includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2069,43 +2077,34 @@ static function() {
 
 				$styles = file_get_contents( $css_file ); //phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
 
-				// Make the rules for the mobile menu less specific so that it can also be applied to the
-				// amp-lightbox mobile menu also.
-				$new_styles = str_replace( '.primary-navigation > .primary-menu-container', '.primary-navigation .primary-menu-container', $styles );
-				$new_styles = str_replace( '.primary-navigation > div > .menu-wrapper', '.primary-navigation div > .menu-wrapper', $new_styles );
-
 				// Append any extra rules that may be needed.
-				$new_styles .= '
-					.primary-navigation-open .menu-button-container {
-						/* Needs to be shown above amp-lightbox, which has a z-index of 1000. */
-						z-index: 1001;
-					}
-
+				$styles .= '
+					/* Trap keyboard navigation within mobile menu when it\'s open */
 					@media only screen and (max-width: 481px) {
-						/* Hide the desktop menu so that it does not interfere when the amo-lightbox mobile menu is open. */
-						.primary-navigation > .primary-menu-container {
-							display: none;
+						.primary-navigation-open #page {
```

Comment:
I just had the idea of doing:

```suggestion
						.primary-navigation-open > * {
```

And that mostly works. It even means that the `.lock-scrolling` style rule could be removed from `MobileRedirection`… _except_ that this then causes the admin bar to be hidden when the menu is opened. So then it could be changed rather to:

```suggestion
						.primary-navigation-open > *:not(#wpadminbar) {
```

But then it causes other elements such as Query Monitor's panel to not appear when the menu is expanded. 

Nevertheless, this doesn't help the non-AMP version. So the style in `MobileRedirection` should remain for now to account for both AMP and non-AMP cases.